go/oasis-node/cmd/debug/txsource/workload/commission.go

```diff
@@ -0,0 +1,251 @@
+package workload
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"time"
+
+	"google.golang.org/grpc"
+
+	"github.com/oasislabs/oasis-core/go/common/crypto/signature"
+	memorySigner "github.com/oasislabs/oasis-core/go/common/crypto/signature/signers/memory"
+	"github.com/oasislabs/oasis-core/go/common/logging"
+	consensus "github.com/oasislabs/oasis-core/go/consensus/api"
+	"github.com/oasislabs/oasis-core/go/consensus/api/transaction"
+	epochtime "github.com/oasislabs/oasis-core/go/epochtime/api"
+	runtimeClient "github.com/oasislabs/oasis-core/go/runtime/client/api"
+	staking "github.com/oasislabs/oasis-core/go/staking/api"
+)
+
+// NameCommission is the name of the commission schedule amendements
+// workload.
+const NameCommission = "commission"
+
+type commission struct {
+	logger *logging.Logger
+
+	rules          staking.CommissionScheduleRules
+	account        signature.Signer
+	reckonedNonce  uint64
+	fundingAccount signature.Signer
+}
+
+func (c *commission) doAmendCommissionSchedule(ctx context.Context, rng *rand.Rand, cnsc consensus.ClientBackend, stakingClient staking.Backend) error {
+	c.logger.Debug("amend commission schedule")
+
+	// Get current epoch.
+	currentEpoch, err := cnsc.GetEpoch(ctx, consensus.HeightLatest)
+	if err != nil {
+		return fmt.Errorf("GetEpoch: %w", err)
+	}
+
+	var account *staking.Account
+	account, err = stakingClient.AccountInfo(ctx, &staking.OwnerQuery{
+		Height: consensus.HeightLatest,
+		Owner:  c.account.Public(),
+	})
+	if err != nil {
+		return fmt.Errorf("stakingClient.AccountInfo %s: %w", c.account.Public(), err)
+	}
+	commissionSchedule := account.Escrow.CommissionSchedule
+
+	// Must not alter bounds before `currentEpoch + RateBoundLead + 1`
+	// interval (+2 since the epoch could have changed before this transacrion
+	// is actually submitted).
```

Comment:
Yes this is already considered.

So this is: `currentEpoch + RateBoundLead + 1 + 1`  (+1 since we cannot amend on the `currentEpoch + RateBoundLead`, and another `+1` to cover the case of epoch changing recently).  

Should update the comment to make it clearer.